src/library/cratehighlightdelegate.cpp

```diff
@@ -0,0 +1,28 @@
+#include "library/cratehighlightdelegate.h"
+#include "library/treeitem.h"
+#include "trackinfoobject.h"
+#include "libraryfeature.h"
+#include "cratefeature.h"
+
+
+void CrateHighlightDelegate::initStyleOption(QStyleOptionViewItem *option,
+                     const QModelIndex &index) const{
+    QStyledItemDelegate::initStyleOption(option, index);
+    QStyleOptionViewItemV4 *optionV4 = qstyleoption_cast<QStyleOptionViewItemV4*>(option);
+
+    TreeItem *item = static_cast<TreeItem*>(index.internalPointer());
+    LibraryFeature* pFeature = static_cast<LibraryFeature*>(item->getFeature());
+    if (pFeature){
+        TrackPointer pTrack = pFeature->getSelectedTrack();
+        if (pTrack){
+            CrateFeature* pCrateFeature = static_cast<CrateFeature*>(pFeature);
```

Comment:
this will != null if the assumption above is valid. I think you want to check if pFeature is a crate first. 
So you might want to use dynamic_cast. 

But in general it is better to avoid all these upcasting. This can be done, by moving the required functions to the base class. Implement them in a generic way and virtual override them in CrateFeature.